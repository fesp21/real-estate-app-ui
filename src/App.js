import React,{ Component, PropTypes } from 'react';
import { StyleSheet } from 'react-native';
import { connect } from "react-redux";
import { StackNavigation, TabNavigation, TabNavigationItem } from '@exponent/ex-navigation';
import Router from './lib/router';
import Ionicons from 'react-native-vector-icons/Ionicons';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import NavBar from './common/NavBar';
import Colors from './common/Colors';

class App extends Component {

  static propTypes = {
    app:PropTypes.object.isRequired
  };

  static route = {
    navigationBar:{
      visible:false
    }
  };

  componentDidMount() {
    // this.props.dispatch(ACTIONS.boot());
  }

  render() {

    const {app,authReducer} = this.props;
    // if(!this.props.app.bootstrapped) return null;


    return (
      <TabNavigation
        id="homeTab"
        navigatorUID="homeTab"
        initialTab="third"
        tabBarStyle={{ backgroundColor:Colors.primary }}
      >
        <TabNavigationItem
          id="first"
          renderIcon={(isSelected) => <Ionicons name="ios-home" size={24} color={isSelected ? 'tomato':'white'}

          /> }>
          <StackNavigation
            id="propertiesStack"
            navigatorUID="propertiesStack"
            initialRoute={Router.getRoute('propertyList')}
            defaultRouteConfig={{
              navigationBar: {
                backgroundColor: 'rgba(255,255,255,0.2)',
                tintColor: '#fff',
                translucent:true,
                renderBackground: (props) => <NavBar {...props} />
              }
            }}
          />
        </TabNavigationItem>

        <TabNavigationItem
          id="second"
          renderIcon={(isSelected) => <FontAwesome name="heart" size={20} color={isSelected ? 'tomato':'white'}
          /> }>
          <StackNavigation
            id="favoritesStack"
            navigatorUID="favoritesStack"
            initialRoute={authReducer.isAuthenticated ? Router.getRoute('propertyFavorites') : Router.getRoute('login',{redirectRoute:'propertyFavorites'}) }
            defaultRouteConfig={{
              navigationBar: {
                backgroundColor: 'rgba(255,255,255,0.2)',
                tintColor: '#fff',
                translucent:true,
                renderBackground: (props) => <NavBar {...props} />
              }
            }}
          />
        </TabNavigationItem>

        <TabNavigationItem
          id="third"
          renderIcon={(isSelected) => <FontAwesome name="plus" size={20} color={isSelected ? 'tomato':'white'}
          /> }>
          <StackNavigation
            id="createStack"
            navigatorUID="createStack"
            initialRoute={Router.getRoute('propertyCreate')}
            defaultRouteConfig={{
              navigationBar: {
                backgroundColor: 'rgba(255,255,255,0.2)',
                tintColor: '#fff',
                translucent:true,
                renderBackground: (props) => <NavBar {...props} />
              }
            }}
          />
        </TabNavigationItem>

        <TabNavigationItem
          id="fourth"
          renderIcon={(isSelected) => <Ionicons name="ios-settings" size={24} color={isSelected ? 'tomato':'white'}
          /> }>
          <StackNavigation
            id="settingsStack"
            navigatorUID="settingsStack"
            initialRoute={Router.getRoute('settingList')}
            defaultRouteConfig={{
              navigationBar: {
                backgroundColor: 'rgba(255,255,255,0.2)',
                tintColor: '#fff',
                translucent:true,
                renderBackground: (props) => <NavBar {...props} />
              }
            }}
          />

        </TabNavigationItem>

      </TabNavigation>
    );

  }
}

function mapStateToProps(state) {
  return {
    app: state.appReducer,
    authReducer:state.authReducer
  }
}

const styles = StyleSheet.create({
  selectedTab: {
    backgroundColor:'green'
  }
});

export default connect(mapStateToProps)(App);