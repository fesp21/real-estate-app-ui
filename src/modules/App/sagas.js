import isNull from 'lodash/isNull';
import { put, call, select, takeLatest } from 'redux-saga/effects';
import { getItem as getStoredItem, setItem } from '../../lib/storage';
import { API as AUTH_API, AUTH_STORAGE_KEY } from '../Auth/api';
import { ACTION_TYPES as AUTH_ACTION_TYPES } from '../Auth/actions';
import { fetchProperties } from '../Property/sagas';
import { ACTION_TYPES } from './actions';
import { COUNTRY_KEY } from './reducer';

function* bootApp() {
  const token = yield call(getStoredItem, AUTH_STORAGE_KEY);
  let currentCountry = yield call(getStoredItem, COUNTRY_KEY);

  if (isNull(currentCountry)) {
    const state = yield select();
    currentCountry = state.appReducer.country;
  }

  try {
    const response = yield call(AUTH_API.login, null, token);
    yield put({ type: AUTH_ACTION_TYPES.LOGIN_SUCCESS, payload: response.data });
  } catch (error) {
    yield put({ type: AUTH_ACTION_TYPES.LOGIN_FAILURE, error });
  }
  yield put({ type: ACTION_TYPES.COUNTRY_CHANGED, country: currentCountry });
  yield put({ type: ACTION_TYPES.BOOT_SUCCESS });
}

function* changeCountrySaga(action) {
  yield call(setItem, COUNTRY_KEY, action.country);
  yield put({ type: ACTION_TYPES.COUNTRY_CHANGED, country: action.country });
  yield call(fetchProperties, action);
}

export function* bootstrapMonitor() {
  yield takeLatest(ACTION_TYPES.BOOT_REQUEST, bootApp);
}

export function* changeCountryMonitor() {
  yield takeLatest(ACTION_TYPES.CHANGE_COUNTRY, changeCountrySaga);
}
