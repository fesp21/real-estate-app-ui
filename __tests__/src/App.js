// __tests__/Intro-test.js
import 'react-native';
import React from 'react';
import App from '../../src/App';
import { Provider } from 'react-redux';
import Store from './../../src/lib/store';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders App Component', () => {
  const tree = renderer.create(

    <Provider store={Store} >
      <App />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});