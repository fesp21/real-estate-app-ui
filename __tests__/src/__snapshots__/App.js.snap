exports[`test renders App Component 1`] = `null`;
